taglib/mpeg/mpegproperties.cpp

```diff
@@ -155,6 +156,25 @@ bool MPEG::Properties::isOriginal() const
 
 void MPEG::Properties::read(File *file)
 {
+  if(file->containerFormat() == File::ADTS) {
+    const MPEG::ADTSHeader firstHeader(file, 0);
+
+    if(firstHeader.bitrate() > 0) {
+      d->bitrate = firstHeader.bitrate();
+      d->length = file->length() / d->bitrate / 8;
+    }
+
+    d->sampleRate        = firstHeader.sampleRate();
+    d->channels          = firstHeader.channelMode() == ADTSHeader::FrontCenterLeftRightSideLeftRightBackLeftRightLFE ? 8 : static_cast<int>(firstHeader.channelMode());
```

Comment:
Also note that TagLib usually tries to keep lines to about 80 characters wide.  This one is 170 wide.

Reply:
turned that into if/else